pennylane/gradients/finite_difference.py

```diff
@@ -359,3 +359,255 @@ def processing_fn(results):
         return qml.math.T(qml.math.stack(grads))
 
     return gradient_tapes, processing_fn
+
+
+@gradient_transform
+def finite_diff_new(

```

Comment:
Hmm, on that note, having `_new` versions means we're duplicating code completely and developers have to be aware of this all along until we keep both versions (which will likely be at least 2 more releases).

There was already a major shortcoming to this that Albert found in #2886: the [hotfix for adjoint computation was only applied to `execute`](https://github.com/PennyLaneAI/pennylane/pull/2886#discussion_r960491539) and not to `execute_new`. If there's a single function that has `if qml.active_return` blocks, such a change would have affected the new return type logic and tests too.

Also, personally, when editing a single file, I do find it more challenging to be able to tell which function I'm editing: the original version or the `_new` one. This is likely editor specific though.

> It'll be a pain to search for all the if qml.active_return blocks, and there's always the possibility of missing some.

To me, this would not be so challenging. A use of `grep` will tell us the occurrences across the entire repo and we'd just have to go through the list.

All in all, I do see the pros/cons of both approaches though. My original question here is more so about the potential cons of mixing these.